stable/cluster-autoscaler/templates/deployment.yaml

```diff
@@ -142,6 +142,13 @@ spec:
             - name: {{ $key }}
               value: "{{ $value }}"
           {{- end }}
+          {{- range $key, $value := .Values.envFromConfigMap }}
```

Comment:
I'm not sure I follow. Does the intended usage look like this? What's the point of using a map and ignoring `$key`?

```yaml
envFromConfigMap:
  ignoredKey:
    name: MY_ENV_VAR
    config_name: my-config-map
    config_key: myEnvVar
```

Reply:
Maybe we could do something like the following instead.

```yaml
{{- range $key, $value := .Values.envFromConfigMap }}
- name: {{ $key }}
  valueFrom:
    configMapKeyRef:
      name: {{ default (include "cluster-autoscaler.fullname" $) $value.name }}
      key: {{ required "Must specify key!" $value.key }}
{{- end }}
```

```yaml
envFromConfigMap:
  MY_ENV_VAR:
    # name: my-config-map
    key: myEnvVar
```

Reply:
Alternatively, I see this pattern used once or twice: https://github.com/helm/charts/blob/4719d54/incubator/common/templates/_envvar.tpl#L9-L19

Reply:
I had this structure, but as environment variable names are unique makes sense to use the form you suggested in your first point. Will amend shortly :)

```
envFromConfigMap:
  - name: MY_ENV_VAR
    config_name: my-config-map
    config_key: myEnvVar
```

Reply:
Updated